Automattic/jetpack
# Premium Blocks: Add banner to block inserter preview

This PR aims to add an upgrade banner to the block inserter preview for Jetpack blocks that require an upgrade.

#### Changes proposed in this Pull Request:

Functional:
* Reactivate block inserter preview for paid blocks and hide the upgrade nudge
* Add a new upgrade banner below the block inserter preview

Technical:
* Refactor existing `withHasWarningIsInteractiveClassNames` HOC to a more generic `withCustomClassNames` HOC to allow any class name to be passed in.
* Make use of `/shared/paid-blocks` file structure that will be added with https://github.com/Automattic/jetpack/pull/16482

|Before|After|
|-|-|
|<img width="697" alt="Screenshot 2020-07-15 at 14 02 05" src="https://user-images.githubusercontent.com/1562646/87542797-d98bad00-c6a3-11ea-9e10-620eafb6c420.png">|<img width="686" alt="Screenshot 2020-07-15 at 14 01 02" src="https://user-images.githubusercontent.com/1562646/87542813-df818e00-c6a3-11ea-996f-39f4a6ab23dd.png">|

Closes https://github.com/Automattic/wp-calypso/issues/44055


#### Testing instructions:






* Be on a free plan that doesn't have access to premium blocks (for quick local testing you can change [this line](https://github.com/Automattic/jetpack/blob/master/extensions/shared/register-jetpack-block.js#L83) to `const requiredPlan = ! requiresPaidPlan( unavailableReason, details );`).
* Open the block inserter
* Hover over a (paid) block
* Observe new upgrade banner in block inserter preview

#### Known issues:
Some blocks have previews that are broken in one form or another (too small, breaking out of preview container, no space below the block, etc.). These issues with the previews of said blocks lead to the banner looking off. In order to get rid of these, the previews of those blocks will have to be fixed.
